src/osp/tasks/top_session.h

```diff
@@ -45,7 +45,12 @@
 #define OSP_AUX_DCDI_A(...) OSP_AUX_DCDI_B(OSP_AUX_DCDI_C(__VA_ARGS__))
 
 /**
- * @brief
+ * @brief Gives the next available section of topData to a session
+ * and declares an ID variable for each element that is given.
+ * The same ID's are also stored inside of the sessions m_data member
+ *
+ * @param count The number of id variables passed to the macro.
+ * @param ids A set of variables that will each be assigned to a unique index of topData.
```

Comment:
some spacing and formatting changes:
```
@brief Gives the next available section of topData to a session
       and declares an ID variable for each element that is given.
       
The same ID's are also stored inside of the sessions m_data member
```
These comments can be used to generate html documentation through doxygen, but I've never actually run it before. I only really care about being able to reading docs in the header code itself, since this is where I'm going to be looking most of the time.

Also, the params here are suppose to be (session, topData, arglist) but it says count and ids? Specifically just the arglist is a count and variable names.

--- extra unneeded information:

There are actually some janky ways of counting the number of arguments instead of passing a number of arguments, but I've yet to implement any. 

I mean jank because there's no way to measure the number of args in __VA_ARGS__. one way I'm thinking of doing this is turning __VA_ARGS__ into a struct of chars or std::byte, then measuring its size:
```c
struct Dummy{
    std::byte __VA_ARGS__; // becomes "std::byte argA, argB, argC, argD;"
};
int argsCount = sizeof(Dummy);
```

Reply:
You could use https://www.boost.org/doc/libs/1_69_0/libs/preprocessor/doc/ref/variadic_size.html

But we don't depend on boost and adding that dependency just for a macro seems like a bad idea.